clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -950,6 +950,9 @@ void CodeGenModule::Release() {
       UsedArray.push_back(llvm::ConstantExpr::getPointerBitCastOrAddrSpaceCast(
           GetAddrOfGlobal(GD), Int8PtrTy));
     }
+    // Sort decls by name to always emit them in deterministic order.
```

Comment:
nit: it looks like the map is `CUDAExternalDeviceDeclODRUsedByHost`. Can that be turned into a `MapVector` instead, which makes any other loops over the map deterministic? More, correct by construction, rather than having to do an extra sort.

Reply:
Thanks for the hint. `MapVector` sounds like the right tool for the job.